tests/ens/registrar/test_bid.py

```diff
@@ -0,0 +1,91 @@
+
+import pytest
+from web3 import Web3
+
+from ens.exceptions import BidTooLow, InvalidLabel, UnderfundedBid
+
+
+@pytest.fixture
+def reg_bid(registrar, mocker):
+    mocker.patch.object(registrar.core, 'shaBid')
```

Comment:
Is there a reason you are mocking this rather than actually letting it issue the transaction.  It seems like issuing the transaction, then pulling the transaction data and verifying that the transaction data is what it's supposed to be would only be marginally more difficult.

This shouldn't even require interacting with the actual contract since all you care about is that the prepared transaction data is as expected.

Reply:
I'm missing something here. What do you mean by "issuing the transaction, then pulling the transaction data and verifying ... without interacting with the actual contract"? Isn't issuing this transaction necessarily attempting to interact with a contract?

Reply:
I'm missing something too because I'm not exactly sure what I was talking about.......  😭